### PR TITLE
feat: add WSGI adapter

### PR DESCRIPTION
**Pull request by WilliamBergamin**
This PR introduced a WSGI adapter to bolt python, this will allow Bolt to be deployed in production without the need for an other WSGI compatible Web Framework like Flask

# Testing

1. Pull this branch
3. Use the `examples/wsgi/app.py` as starting point for a test app
4. Run `scripts/build_pypi_package.sh`
5. Use the `.whl` file found under `dist/` to run `pip install /path/to/dist/slack_bolt.whl`
6. Run `pip install gunicorn`
7. Run `gunicorn app:api -b 0.0.0.0:3000 --log-level debug` to start the application

### Category

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [x] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`
* [ ] Others

## Requirements 

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

**Comment by seratch** [comment]
Can you check the flake8 errors? https://github.com/slackapi/bolt-python/actions/runs/9228875341/job/25393855815?pr=1085

File: slack_bolt/adapter/wsgi/utils.py [hunk 1]
```diff
@@ -0,0 +1 @@
+ENCODING = "utf-8"  # should always be utf-8
```
Comment (seratch) [inline slack_bolt/adapter/wsgi/utils.py:1]: This is fine for Slack apps but perhaps the comment should be clearer to clarify we understand this is not true for WSGI in general:
```suggestion
ENCODING = "utf-8"  # The content encoding for Slack requests/responses is always utf-8
```

Comment (seratch) [reply slack_bolt/adapter/wsgi/utils.py:1]: Do you think it's good to rename this file to "intenrals.py" for consistency with other source files in this project?

**Review by seratch — COMMENTED** [review]
Thanks for working on this. This is a nice addition! Once you clean up things, we can go ahead with merging the change for the next minor release.

File: slack_bolt/adapter/wsgi/utils.py (hunk 1 above)
Comment (WilliamBergamin) [reply slack_bolt/adapter/wsgi/utils.py]: Renamed the file to `internals.py` and updated the comment 💯

File: slack_bolt/adapter/wsgi/http_request.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        self.environ = environ
+
+    def get_headers(self) -> Dict[str, str]:
+        headers = {}
+        for key, value in self.environ.items():
+            if key in {"CONTENT_LENGTH", "CONTENT_TYPE"}:
+                name = key.lower().replace("_", "-")
+                headers[name] = value
+            if key.startswith("HTTP_"):
+                name = key[len("HTTP_"):].lower().replace("_", "-")  # fmt: skip
```
Comment (WilliamBergamin) [inline slack_bolt/adapter/wsgi/http_request.py:29]: Turns out the black formatter was cause the Flake8 [issue](https://github.com/psf/black/issues/1323), the workaround if to add `# fmt: skip` on the line causing the issue

Comment (WilliamBergamin) [reply slack_bolt/adapter/wsgi/http_request.py:29]: The Black Formatter will add a whitespace between `len("HTTP_")` and `:`, Flake8 does not align with this syntax, the Black Formatter does not allow ignoring specific rules

`key[len("HTTP_") :].lower().replace("_", "-")`

Comment (seratch) [reply slack_bolt/adapter/wsgi/http_request.py:29]: this makes sense; thanks for clarifying it

**Review by seratch — APPROVED** [review]
thank you; looks great to me :+1:

**Pull request merged**